Core/include/Acts/Utilities/Zip.hpp

```diff
@@ -21,16 +21,16 @@ namespace Acts {
 /// @param rb The second range
 /// @note the behaviour is undefined if the ranges do not have equal range
 template <typename RA, typename RB>
-auto zip(RA &ra, RB &rb) {
+auto zip(RA &&ra, RB &&rb) {
   using ItA = decltype(ra.begin());
   using ItB = decltype(rb.begin());
 
   struct It {
     ItA a;
     ItB b;
 
-    using reference =
-        std::tuple<typename ItA::reference, typename ItB::reference>;
+    using reference = std::tuple<decltype(*std::declval<ItA>()),
+                                 decltype(*std::declval<ItB>())>;
```

Comment:
I needed this to make it work with the proxy iterators. Do you see any issues @asalzburger?

Reply:
Should be ok.